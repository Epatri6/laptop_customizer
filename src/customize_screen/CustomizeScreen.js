import React from 'react';
import './CustomizeScreen.css';
import FeatureSelection from '../feature_selection/FeatureSelection';

export default class CustomizeScreen extends React.Component {
  listFeatures = () => {
    const {features, selectedFeatures, updateFeature} = this.props;
    return Object.keys(features).map((feature, idx) => {
      const featureHash = feature + '-' + idx;
      return <FeatureSelection key={featureHash} id={featureHash} feature={features[feature]} featureName={feature} selectedFeature={selectedFeatures[feature]} updateFeature={updateFeature}/>
    });
  };

  render() {
    return (
      <form className="main__form">
        <h2>Customize your laptop</h2>
        {this.listFeatures()}
      </form>
    );
  }

  static defaultProps = {
    features: {},
    selectedFeatures: {},
    updateFeature: () => {}
  }
}
