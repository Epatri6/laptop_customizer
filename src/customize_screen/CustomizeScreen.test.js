import React from 'react';
import ReactDOM from 'react-dom';
import CustomizeScreen from './CustomizeScreen';
import renderer from 'react-test-renderer';

describe('Testing ShoppingCart Component', () => {

  const testProps = {
    features: {
      Processor: [
        {
          name: '17th Generation Intel Core HB (7 Core with donut spare)',
          cost: 700
        },
        {
          name: 'Professor X AMD Fire Breather with sidewinder technology',
          cost: 1200
        }
      ],
      "Operating System": [
        {
          name: 'Ubuntu Linux 16.04',
          cost: 200
        },
        {
          name: 'Bodhi Linux',
          cost: 300
        }
      ]
    },
    selectedFeatures: {
      Processor: {
        name: '17th Generation Intel Core HB (7 Core with donut spare)',
        cost: 700
      },
      'Operating System': {
        name: 'Ubuntu Linux 16.04',
        cost: 200
      }
    },
    updateFeature: () => {}
  };

  const {features, selectedFeatures, updateFeature} = testProps;

  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CustomizeScreen features={features} selectedFeatures={selectedFeatures} updateFeature={updateFeature}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CustomizeScreen />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders as expected', () => {
    const tree = renderer
      .create(<CustomizeScreen features={features} selectedFeatures={selectedFeatures} updateFeature={updateFeature}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  
  it('renders as expected without props', () => {
    const tree = renderer
      .create(<CustomizeScreen />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});