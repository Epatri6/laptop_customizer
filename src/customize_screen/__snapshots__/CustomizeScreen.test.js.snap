// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ShoppingCart Component renders as expected 1`] = `
<form
  className="main__form"
>
  <h2>
    Customize your laptop
  </h2>
  <fieldset
    className="feature"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Processor
      </h3>
    </legend>
    <div
      className="feature__item"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
        name="Processor"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
      >
        17th Generation Intel Core HB (7 Core with donut spare)
         (
        $700.00
        )
      </label>
    </div>
    <div
      className="feature__item"
    >
      <input
        checked={false}
        className="feature__option"
        id="\\"name\\":\\"Professor-X-AMD-Fire-Breather-with-sidewinder-technology\\"\\"cost\\":1200"
        name="Processor"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Professor-X-AMD-Fire-Breather-with-sidewinder-technology\\"\\"cost\\":1200"
      >
        Professor X AMD Fire Breather with sidewinder technology
         (
        $1,200.00
        )
      </label>
    </div>
  </fieldset>
  <fieldset
    className="feature"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Operating System
      </h3>
    </legend>
    <div
      className="feature__item"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"Ubuntu-Linux-16.04\\"\\"cost\\":200"
        name="Operating-System"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Ubuntu-Linux-16.04\\"\\"cost\\":200"
      >
        Ubuntu Linux 16.04
         (
        $200.00
        )
      </label>
    </div>
    <div
      className="feature__item"
    >
      <input
        checked={false}
        className="feature__option"
        id="\\"name\\":\\"Bodhi-Linux\\"\\"cost\\":300"
        name="Operating-System"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Bodhi-Linux\\"\\"cost\\":300"
      >
        Bodhi Linux
         (
        $300.00
        )
      </label>
    </div>
  </fieldset>
</form>
`;

exports[`Testing ShoppingCart Component renders as expected without props 1`] = `
<form
  className="main__form"
>
  <h2>
    Customize your laptop
  </h2>
</form>
`;
